src/pokenav_region_map.c

```diff
@@ -740,3 +769,23 @@ static void SetCityZoomTextInvisibility(bool32 invisible)
     for (i = 0; i < (int)ARRAY_COUNT(state->cityZoomTextSprites); i++)
         state->cityZoomTextSprites[i]->invisible = invisible;
 }
+
+void UpdateRegionMapHelpBarText(void)
+{
+    struct RegionMap* regionMap = GetSubstructPtr(POKENAV_SUBSTRUCT_REGION_MAP);
+
+    if (regionMap->mapSecType == MAPSECTYPE_CITY_CANFLY && OW_FLAG_POKE_RIDER)
```

Comment:
```suggestion
    if (regionMap->mapSecType == MAPSECTYPE_CITY_CANFLY && OW_FLAG_POKE_RIDER && Overworld_MapTypeAllowsTeleportAndFly(gMapHeader.mapType))
```
Handling for prevention of flying indoors etc.